GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -62,11 +63,11 @@ static GitProcess()
         }
 
         public GitProcess(Enlistment enlistment)
-            : this(enlistment.GitBinPath, enlistment.WorkingDirectoryRoot, enlistment.GVFSHooksRoot)
+            : this(enlistment.GitBinPath, enlistment.WorkingDirectoryRoot, enlistment.GVFSHooksRoot, enlistment.GitObjectsRoot)
         {
         }
 
-        public GitProcess(string gitBinPath, string workingDirectoryRoot, string gvfsHooksRoot)
+        public GitProcess(string gitBinPath, string workingDirectoryRoot, string gvfsHooksRoot, string gitObjectsRoot = null)

```

Comment:
I don't love having an optional param here.  Perhaps I should force the callers to send in null.  Let me know if there is a more preferred way to do this.

Since we use gitObjectRoot in the Environment vars, i think it should be passed in to the constructor (similar ot workingDirectoryRoot)

Reply:
hm. strange we can't do it during `PrunePacked()`. It certainly seems we could insert the variable there and then remove the variable after calling the invoke method. Maybe instead we should extend one of the `InvokeGitAgainstDotGitFolder()` methods?

Reply:
@derrickstolee that's basically what I was debating. :-)

Given we already have variables on the constructor that are used in the environment (and the value here doesn't change) having it here seems like a good option.  But I don't love it being an optional param, I could go and update the callers to pass in null.

Alternatively we could set it on PrunePacked, but that doesn't follow how the class handles env variables currently.  

@wilbaker  would love your input here too1